pcsx2/DEV9/ATA/HddCreate.cpp

```diff
@@ -81,32 +86,102 @@ void HddCreate::WriteImage(ghc::filesystem::path hddPath, int reqSizeMiB)
 		return;
 	}
 
+	//Create File
 	std::fstream newImage = ghc::filesystem::fstream(hddPath, std::ios::out | std::ios::binary);
-
 	if (newImage.fail())
 	{
 		SetError();
 		return;
 	}
+	newImage.close();
 
-	//Size file
-	newImage.seekp(((u64)reqSizeMiB) * 1024 * 1024 - 1, std::ios::beg);
-	const char zero = 0;
-	newImage.write(&zero, 1);
+	//Size file (Slow on FS that don't support sparse)
+	std::error_code ec;
+	ghc::filesystem::resize_file(hddPath, ((u64)fileMiB) * 1024 * 1024, ec);
```

Comment:
Looking at the ghc::filesystem source, it calls ``SetEndOfFile()``.
``SetEndOfFile()`` states that the contents of the extended part of the file is undefined.

On NTFS, extending a file reserves disk space but does not zero out the data immediately, instead It will do that lazily on demand, see https://devblogs.microsoft.com/oldnewthing/20110922-00/?p=9573 and https://devblogs.microsoft.com/oldnewthing/20150710-00/?p=45171.

Reply:
Maybe instead we should set it sparse before resizing?